content/integrations/awslambda.md

```diff
@@ -21,27 +21,93 @@ If you haven't already, set up the [Amazon Web Services integration first](/inte
 In the Amazon Web Services integration tile, ensure that Lambda is checked under metric collection.
 
 To send custom metrics to Datadog, you must print a log line from your Lambda, using the following format:
-`MONITORING|unix_epoch_timestamp|value|metric_type|my.metric.name|#tag1:value,tag2`
 
-Please ensure the `unix_epoch_timestamp` is in seconds (not milliseconds).
+~~~
+MONITORING|unix_epoch_timestamp|value|metric_type|metric.name|#tag1:value,tag2
+~~~
+
+## Notes on each section
+
+### MONITORING
+* This is used to find the log statement within Cloudwatch Logs and pull it into Datadog
+
+### unix_epoch_timestamp
+* Please ensure the `unix_epoch_timestamp` is in seconds (not milliseconds)
+
+### value
+* The value of the metric **must** be a number.
+* For metric_type `check` the options are, `'0': OK, '1': WARNING, '2': CRITICAL, '3': UNKNOWN`
+
+### metric_type
+* Supported metric types are `count, gauge, histogram, and check`
+
+### metric.name
+* A unique name to identify your metric
 
-For example, here is sample snippet for printing a valid custom metric, from your Lambda function (in Python):
+### tags
+* Optionally add tags to apply to this metric
+* If you do not want to add any tags use the format: `MONITORING|unix_epoch_timestamp|value|metric_type|metric.name`
+* The tag `function_name` will automatically be applied to custom metrics
+
+
+## Sample snippets (in Python):
+
+### Count/Gauge
 
 ~~~
 unix_epoch_timestamp = int(time.time())
 value = 42
 metric_type = 'count'
+#metric_type = 'gauge'
```

Comment:
I'm not sure it needs to be pointed out that gauges and counters will have similar looking code.

In fact, the `print` line looks the same for histograms and service checks. The only thing that's different for histogram is the looping over a range to show that a histogram is an aggregation of values.